packages/connect/src/device/__tests__/checkFirmwareRevision.test.ts

```diff
@@ -19,42 +19,39 @@ const ONLINE_RELEASES_JSON_MOCK: FirmwareRelease[] = [
     },
 ];
 
-const createDevicePrams = (
-    params: Partial<CheckFirmwareRevisionParams>,
-): CheckFirmwareRevisionParams => ({
+const DeviceNames = Object.values(DeviceModelInternal);
+
+type CreateDeviceParams = Omit<CheckFirmwareRevisionParams, 'internalModel'>;
+
+const createDeviceParams = (params: Partial<CreateDeviceParams>): CreateDeviceParams => ({
     deviceRevision: '1eb0eb9d91b092e571aac63db4ebff2a07fd8a1f',
     firmwareVersion: [2, 7, 2],
     expectedRevision: '1eb0eb9d91b092e571aac63db4ebff2a07fd8a1f',
-    internalModel: DeviceModelInternal.T1B1,
     ...params,
 });
 
-describe(checkFirmwareRevision.name, () => {
+describe.each(DeviceNames)(`${checkFirmwareRevision.name} for device %s`, internalModel => {
     it.each<{
         it: string;
         httpRequestMock?: () => Promise<FirmwareRelease[]>;
-        params: CheckFirmwareRevisionParams;
+        params: CreateDeviceParams;
         expected: FirmwareRevisionCheckResult;
     }>([
         {
             it: 'passes when firmware revision is same as in static file',
-            params: createDevicePrams({
-                deviceRevision: '1eb0eb9d91b092e571aac63db4ebff2a07fd8a1f',
-                expectedRevision: '1eb0eb9d91b092e571aac63db4ebff2a07fd8a1f',
```

Comment:
since `createDeviceParams` assigns these as default values, I think it's needless to repeat them here, so I removed it, but I can put it back if you'd like to.